components/App.js

```diff
@@ -42,6 +53,98 @@ export const onclickBindings = {
       menu,
     });
   },
+  'threeD-social-tradeBtn': (e) => {
+    const { menu } = getState();
+    menu.trade.visible = true;
+    menu.trade.toPeer = e.name;
+    menu.trade.fromPeer = loginManager.getAddress();
+    setState({
+      menu,
+    });
+  },
+  'threeD-trade-agreement': (e) => {
+    const { menu } = getState();
+    menu.trade.agreement = !menu.trade.agreement;
+    setState({ menu });
+  },
+  'threeD-trade-header-backBtn': (e) => {
+    const { menu } = getState();
+    menu.trade = JSON.parse(JSON.stringify(defaultState.menu.trade));
```

Comment:
This could be abstracted in `clone` util method.

Reply:
e90b9bb